src/main/java/org/terasology/spawning/SpawningAuthoritySystem.java

```diff
@@ -0,0 +1,171 @@
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.inventory.InventoryComponent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.network.NetworkComponent;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.MersenneRandom;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.items.BlockItemComponent;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class SpawningAuthoritySystem extends BaseComponentSystem {
+    private static final Logger logger = LoggerFactory.getLogger(SpawningAuthoritySystem.class);
+
+    @In
+    private BlockManager blockManager;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private InventoryManager inventoryManager;
+
+    private Prefab prefabToSpawn;
+    private Vector3f spawnPos;
+
+
+    /**
+     * Spawns the desired Oreon at the location of Portal which sends the event
+     */
+    @ReceiveEvent
+    public void oreonSpawn(OreonSpawnEvent event, EntityRef player) {
+        prefabToSpawn = event.getOreonPrefab();
+        spawnPos = event.getSpawnPos();
+        spawnPos.y = spawnPos.y - (float)0.5;
+
+        boolean toSpawn = consumeItem(player, prefabToSpawn);
+        if(toSpawn) {
+            // spawn the new oreon into the world
+            EntityRef newOreon = entityManager.create(prefabToSpawn, spawnPos);
+            NetworkComponent netComp = new NetworkComponent();
+            netComp.replicateMode = NetworkComponent.ReplicateMode.ALWAYS;
+            newOreon.addComponent(netComp);
+            newOreon.getComponent(OreonSpawnComponent.class).parent = player;
+
+            assignRandomAttributes(newOreon);
+
+            logger.info("Player " + newOreon.getComponent(OreonSpawnComponent.class).parent + "Spawned a new Oreon of Type : " + prefabToSpawn);
+        }
+    }
+
+    public boolean consumeItem(EntityRef player, Prefab prefab) {
+        OreonSpawnComponent oreonSpawnComponent = prefab.getComponent(OreonSpawnComponent.class);
+        HashMap<String, Integer> itemsHashMap = oreonSpawnComponent.itemsToConsume;
+
+        if (oreonSpawnComponent.itemsToConsume != null) {
+            ArrayList<String> neededItemList = new ArrayList(itemsHashMap.keySet());
+            int numberOfItems = neededItemList.size();
+            if (player.hasComponent(InventoryComponent.class)) {
+                ArrayList<Integer> requiredSlots = getSlotsForRequiredItems(itemsHashMap, player);
+
+                //all required items not in inventory
+                if (requiredSlots.size() != numberOfItems) {
+                    logger.info("Could not find all required items in inventory");
```

Comment:
Implementing this would require a major change in the logic of reduction of resources or maintain a list of resources that have been found, and then log all the remaining resources.